### PR TITLE
CB-10444 - fixed error message for wrong osType when validation a …

### PR DESCRIPTION
**Pull request by attilapalfi92**
… HDP repository json in maintenance mode. fix the uploading / saving mechanism of that json (StackEndpoint / put)

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java [hunk 1]
```diff
@@ -371,12 +374,24 @@ public String getStackRepositoryJson(Long stackId) {
...
+            String stackRepositoryJson = ambariClient.getLatestStackRepositoryAsJson(cluster.getName(), osType, stackRepoId);
+            if (stackRepositoryJson == null) {
+                throw new BadRequestException(String.format("Stack Repository response coming from Ambari server was null "
+                        + "for cluster '%s' and repo url '%s'.", cluster.getName(), stackRepoId));
+            }
+            return stackRepositoryJson;
+        } catch (Exception e) {
+            // this instanceof is necessary because the groovy client hides somehow the HttpResponseException
+            // see https://jira.cloudera.com/browse/CB-10444 for more information
+            if (e instanceof HttpResponseException) {
```
Comment (pdarvasi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java:387]: so you mean HttpResponseException cannot be caught separately?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java [hunk 2]
```diff
@@ -371,12 +374,24 @@ public String getStackRepositoryJson(Long stackId) {
...
+                HttpResponseException cast = HttpResponseException.class.cast(e);
+                if (cast.getResponse().getStatus() == HttpStatus.SC_NOT_FOUND) {
+                    throw new BadRequestException("The specified osType is wrong. " + cast.getMessage());
```
Comment (pdarvasi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java:390]: pls. add osType value to the logline, too

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java (hunk 1 above)
Comment (attilapalfi92) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java:387]: java compiler doens't let you catch a checked exception if its not listed in the method signature. thats the point of doing this weird implementation.

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java (hunk 2 above)
Comment (attilapalfi92) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterService.java:390]: ok

**Review by sodre90 — APPROVED** [review]

**Review by pdarvasi — APPROVED** [review]

**Pull request merged**